src/foam/u2/crunch/wizardflow/CreateWizardletsAgent.js

```diff
@@ -38,55 +38,70 @@ foam.CLASS({
   ],
 
   methods: [
-    function execute() {
-      return Promise.all(this.capabilities
-        .reduce((updateUCJPromiseList, cap) => {
-          var associatedEntity, wizardlet;
-            
-          if ( Array.isArray(cap) && ( foam.nanos.crunch.MinMaxCapability.isInstance(cap[cap.length - 1]) ) ){
-            var minMaxCap = cap[cap.length - 1];
-
-            minMaxArray = [];
-
-            var choiceWizardlets = cap.slice(0, cap.length - 1).map(
-              capability => {
-                wizardlet = capability.wizardlet.cls_.create(
-                { 
-                  capability: capability, 
-                  isAvailable: false,
-                  ...capability.wizardlet.instance_ 
-                }, this)
+    function parseArrayToWizards(array, parentWizardlet){
+      var isOr = foam.nanos.crunch.MinMaxCapability.isInstance(array[array.length - 1]) ? true : false;
+      var updateUcjPromiseList = [];
 
-                associatedEntity = capability.associatedEntity === foam.nanos.crunch.AssociatedEntity.USER ? this.subject.user : this.subject.realUser;
+      var currentCap = array[array.length - 1];
+      var currentWizardlet  = isOr 
+        ? foam.nanos.crunch.ui.MinMaxCapabilityWizardlet.create({
```

Comment:
same as below

Reply:
Done. Don't even  need the ternary operator anymore if  I override 'wizardlet' property on MinMaxCapability to create  a MinMaxCapabilityWizardlet in the factory.